### ISSUE TITLE
RuboCop does not trigger `Security/Open` in `Rakefile`

### ISSUE THREAD

**Issue by lopopolo** [rubocop/rubocop#10249]
RuboCop does not fire the `Security/Open` lint in a Rakefile that uses `Kernel#open` on a URL.

# Output

```console
$ bundle exec rubocop
Inspecting 2 files
..

2 files inspected, no offenses detected
```

## Debug

```console
$ bundle exec rubocop -d
For /Users/lopopolo/dev/artichoke/rcrake: configuration from /Users/lopopolo/dev/artichoke/rcrake/.rubocop.yml
configuration from /usr/local/var/rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /usr/local/var/rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-rake-0.6.0/config/default.yml
Default configuration from /usr/local/var/rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.22.3/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /Users/lopopolo/.cache/rubocop_cache/dfdb8d442bd33f2ec8d8fd70d677c5f4b21924a9/6d7a3b621ca1730e04accd938619e4bdab66cfb1/bcee1113255ef2cdca85bb20feae52f44c20d800
Loading cache from /Users/lopopolo/.cache/rubocop_cache/dfdb8d442bd33f2ec8d8fd70d677c5f4b21924a9/6d7a3b621ca1730e04accd938619e4bdab66cfb1/ae8d2cfe487bd4e1c273e1b35c537eb311e1817a
Inspecting 2 files
Scanning /Users/lopopolo/dev/artichoke/rcrake/Gemfile
Loading cache from /Users/lopopolo/.cache/rubocop_cache/dfdb8d442bd33f2ec8d8fd70d677c5f4b21924a9/6d7a3b621ca1730e04accd938619e4bdab66cfb1/bcee1113255ef2cdca85bb20feae52f44c20d800
.Scanning /Users/lopopolo/dev/artichoke/rcrake/Rakefile
Loading cache from /Users/lopopolo/.cache/rubocop_cache/dfdb8d442bd33f2ec8d8fd70d677c5f4b21924a9/6d7a3b621ca1730e04accd938619e4bdab66cfb1/ae8d2cfe487bd4e1c273e1b35c537eb311e1817a
.

2 files inspected, no offenses detected
Finished in 0.26521100001991726 seconds
```

```console
$ bundle exec rubocop -C false -d
For /Users/lopopolo/dev/artichoke/rcrake: configuration from /Users/lopopolo/dev/artichoke/rcrake/.rubocop.yml
configuration from /usr/local/var/rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /usr/local/var/rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-rake-0.6.0/config/default.yml
Default configuration from /usr/local/var/rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.22.3/config/default.yml
Inspecting 2 files
Scanning /Users/lopopolo/dev/artichoke/rcrake/Gemfile
.Scanning /Users/lopopolo/dev/artichoke/rcrake/Rakefile
.

2 files inspected, no offenses detected
Finished in 0.17393299998366274 seconds
```

# Meta

```console
$ bundle exec rubocop --verbose-version
1.22.3 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 3.0.2 x86_64-darwin20)
  - rubocop-rake 0.6.0
```

# Setup

In a directory with the following sources:

## `Gemfile`

```ruby
# frozen_string_literal: true

source 'https://rubygems.org'

gem 'rake', '>= 12.3.3', require: false
gem 'rubocop', '~> 1.22', require: false
gem 'rubocop-rake', '~> 0.6', require: false
```

## `Gemfile.lock`

```
GEM
  remote: https://rubygems.org/
  specs:
    ast (2.4.2)
    parallel (1.21.0)
    parser (3.0.2.0)
      ast (~> 2.4.1)
    rainbow (3.0.0)
    rake (13.0.6)
    regexp_parser (2.1.1)
    rexml (3.2.5)
    rubocop (1.22.3)
      parallel (~> 1.10)
      parser (>= 3.0.0.0)
      rainbow (>= 2.2.2, < 4.0)
      regexp_parser (>= 1.8, < 3.0)
      rexml
      rubocop-ast (>= 1.12.0, < 2.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 1.4.0, < 3.0)
    rubocop-ast (1.12.0)
      parser (>= 3.0.1.1)
    rubocop-rake (0.6.0)
      rubocop (~> 1.0)
    ruby-progressbar (1.11.0)
    unicode-display_width (2.1.0)

PLATFORMS
  ruby

DEPENDENCIES
  rake (>= 12.3.3)
  rubocop (~> 1.22)
  rubocop-rake (~> 0.6)

BUNDLED WITH
   2.2.22
```

## `.rubocop.yml`

```yaml
---
require: rubocop-rake
AllCops:
  NewCops: enable
  TargetRubyVersion: 2.6
  DisplayCopNames: true
  Exclude:
    - "**/node_modules/**/*"
    - "**/target/**/*"
    - "**/vendor/**/*"
Layout/EndOfLine:
  EnforcedStyle: lf
Layout/LineLength:
  Enabled: true
  Exclude:
    - "scripts/gen_case_lookups.rb"
    - "**/Rakefile"
Metrics:
  Enabled: false
Security:
  Enabled: true
Style/Documentation:
  Enabled: false
```

## `Rakefile`

```ruby
# frozen_string_literal: true

require 'open-uri'

namespace :unicode do
  desc 'Rebuild Rust generated Rust sources from Unicode data'
  task :build do
    ruby 'scripts/gen_case_lookups.rb'
  end

  desc 'Update Unicode data'
  task :update do
    open('https://www.unicode.org/Public/UCD/latest/ucd/CaseFolding.txt') do |data|
      File.open('CaseFolding.txt', 'w') do |file|
        data.each_line do |line|
          file.write(line)
        end
      end
    end
  end
end
```

**Comment by dvandersluis** [comment]
This isn't a rake, or Rakefile, issue, but rather that `Security/Open` isn't registering for `open` calls with a block. I'll fix it up now.

**Comment by dvandersluis** [comment]
Actually my mistake, the cop isn't registering because it's not considered an offense when given a literal string. In your file, you're giving it a string to open, which is why no offense is registered.

See: https://docs.rubocop.org/rubocop/cops_security.html#securityopen

**Comment by lopopolo** [comment]
Thanks for taking a look @dvandersluis 

I appreciate the changes to the docs to address my confusion.
